MS5607Driver/MS5607Driver.cpp

```diff
@@ -0,0 +1,212 @@
+/** ********************************************************************************
+ * @file    MS5607Driver.cpp
+ * @author  Noah
+ * @date    Nov 16, 2024
+ * @brief	This driver gets pressure and temperature readings from the MS5607 barometers.
+ ******************************************************************************** */
+/************************************ * INCLUDES ************************************/
+#include "MS5607Driver.hpp"
+/************************************ * PRIVATE MACROS AND DEFINES ************************************/
+/************************************ * VARIABLES ************************************/
+constexpr int TEMP_LOW = 2000;
+constexpr int TEMP_VERY_LOW = -1500;
+constexpr int CMD_SIZE = 1;
+constexpr int CMD_TIMEOUT = 150;
+
+// Barometer Commands (should not be modified, non-const due to HAL and C++ strictness)
+static uint8_t ADC_D1_512_CONV_CMD = 0x42;
+static uint8_t ADC_D2_512_CONV_CMD = 0x52;
+static uint8_t ADC_READ_CMD = 0x00;
+static uint8_t PROM_READ_SENS_CMD = 0xA2; // make sure these are the correct commands vv
+static uint8_t PROM_READ_OFF_CMD = 0xA4;
+static uint8_t PROM_READ_TCS_CMD = 0xA6;
+static uint8_t PROM_READ_TCO_CMD = 0xA8;
+static uint8_t PROM_READ_TREF_CMD = 0xAA;
+static uint8_t PROM_READ_TEMPSENS_CMD = 0xAC; // ----------------- ^^
+static uint8_t READ_BYTE_CMD = 0x00;
+static uint8_t RESET_CMD = 0x1E;
+/************************************ * FUNCTION DECLARATIONS ************************************/
+/************************************ * FUNCTION DEFINITIONS ************************************/
+
+/**
+ * @brief gets a single sample of barometer data
+ * @returns a barometer data structure consisting of a 'temp' and 'pressure' variable
+ */
+MS5607_DATA_t MS5607_Driver::getSample(){
```

Comment:
This function could be split into multiple functions to improve readability rather than relying on comments.